check-encryption-leak:misc: refactoring and comments

smagnani96 opened the pull request:
```release-note
Refactoring and code comments for the check-encryption-leak script.
```

pchaigno commented on .github/actions/bpftrace/scripts/check-encryption-leaks.bt:
```diff
@@ -1,16 +1,34 @@
 // $1: IPv4 CiliumInternalIP - Node1
 // $2: IPv6 CiliumInternalIP - Node1
 // $3: IPv4 CiliumInternalIP - Node2
 // $4: IPv6 CiliumInternalIP - Node2
```
Citing myself:

![image](https://github.com/user-attachments/assets/b22eb8cd-4294-4779-bb8d-23a730c6d602)

I think the larger point is that, as a reviewer, it makes things easier because I only have one thing to keep in mind when reading the code, not six :grin:

smagnani96 replied:
Yup, will wait for fixes to get-in, rebase, then refactor this into smaller commits. Thanks :pray:

smagnani96 committed:
check-encryption-leak:misc: refactor and code comments

This commit extends comments in the code to cover all functions/probes
and explanations of our assumptions to execute the script.
In addition, it refactors the current macros CIDR4 and CIDR6 to use the
provided IPv4/6 addresses for CIDR computation. We always assume a /8,
but with this improvement the script can be tested even in env where
PodCIDRs are different than 10.0.0.0/8 or fd00::/8.

Signed-off-by: Simone Magnani <simone.magnani@isovalent.com>

smagnani96 committed:
check-encryption-leak:misc: args checking probe

This commit adds the BEGN probe to the bpftrace script to check arguments
number and correctness.

Signed-off-by: Simone Magnani <simone.magnani@isovalent.com>
